### PR TITLE
Added history tab

### PR DESCRIPTION
**Pull request by okanji**
- Removed history from the dashboard
- History is now shown for an individual project (not for all of them)
- Added a history tab: 
image
- Changed file structure to conditionally render decision or edit pages depending on if an active request exists
- Added history page: 
image
- Small formatting and spacing changes
- Added crumbs nav on top of every page

### CODE REVIEW

File: app/(product)/private-cloud/(product)/product/[licencePlate]/@decision/page.tsx [hunk 1]
```diff
@@ -57,6 +57,12 @@ export default function RequestDecision({ params }: { params: { licencePlate: st
     values: { humanCommnet: '', decision: '', ...data?.requestedProject },
   });
 
+  useEffect(() => {
+    if (!session?.user?.roles?.includes('admin')) {
```
Comment (junminahn) [inline app/(product)/private-cloud/(product)/product/[licencePlate]/@decision/page.tsx:61]: We we use `session?.isAdmin` as it is available now?

File: app/(product)/private-cloud/(product)/product/[licencePlate]/default.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+const Page = () => {
+  return (
+    <div>
+      <h1>Welcome to My React Page</h1>
+    </div>
+  );
+};
+
+export default Page;
```
Comment (junminahn) [inline app/(product)/private-cloud/(product)/product/[licencePlate]/default.tsx:11]: Is this a test page?

File: app/(product)/private-cloud/(product)/product/[licencePlate]/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  decision: React.ReactNode;
+}) {
+  const data = await prisma.privateCloudRequest.findFirst({
+    where: {
+      licencePlate: params.licencePlate,
+      active: true,
+    },
+  });
+
+  return <div className="mt-6">{!data ? edit : decision}</div>;
```
Comment (junminahn) [inline app/(product)/private-cloud/(product)/product/[licencePlate]/layout.tsx:19]: I wonder if `layout` page requires `children` to pass into the pages?

File: app/(product)/public-cloud/(product)/history/page.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+function MyComponent() {
+  return (
+    <div>
+      <h1>Hello, World!</h1>
+    </div>
+  );
+}
+
+export default MyComponent;
```
Comment (junminahn) [inline app/(product)/public-cloud/(product)/history/page.tsx:11]: Is this another test page?

File: app/(product)/public-cloud/(product)/product/[licencePlate]/@decision/page.tsx [hunk 5]
```diff
@@ -67,6 +64,12 @@ export default function RequestDecision({ params }: { params: { licencePlate: st
     }
   }, [data]);
 
+  useEffect(() => {
+    if (!session?.user?.roles?.includes('admin')) {
```
Comment (junminahn) [inline app/(product)/public-cloud/(product)/product/[licencePlate]/@decision/page.tsx:68]: We we use `session?.isAdmin` as it is available now?

File: components/tabs/ProductsRequestsTabs.tsx [hunk 6]
```diff
@@ -13,22 +13,22 @@ export default function Tabs() {
...
-          ${
-            pathname.split('/')[2] === 'products' ? 'bg-bcblue text-white' : 'bg-white text-gray-900 hover:bg-gray-100'
-          }`}
+          ${pathname.split('/')[3] === 'all' ? 'bg-bcblue text-white' : 'bg-white text-gray-900 hover:bg-gray-100'}`}
       >
         Products
       </Link>
       <Link
-        style={{ width: 97 }}
+        style={{ width: 114 }}
```
Comment (junminahn) [inline components/tabs/ProductsRequestsTabs.tsx:21]: Can the `width` be applied as in a `tailwind class`?

**Review by junminahn — COMMENTED** [review]
I see a multiple test pages, are they added as placeholders for now?

File: app/(product)/private-cloud/(product)/product/[licencePlate]/layout.tsx (hunk 3 above)
Comment (okanji) [reply app/(product)/private-cloud/(product)/product/[licencePlate]/layout.tsx:19]: This is using nextjs parallel routes to conditionally render either the edit or decision page depending on if a active request exists. It's the only way I could find to do this but I think it works nicely. In this case you don't need to render the children but instead the edit and decision pages are passed as props.

File: app/(product)/private-cloud/(product)/product/[licencePlate]/default.tsx (hunk 2 above)
Comment (okanji) [reply app/(product)/private-cloud/(product)/product/[licencePlate]/default.tsx:11]: This is required for next js parallel routes, they require a default page

File: app/(product)/private-cloud/(product)/product/[licencePlate]/@decision/page.tsx (hunk 1 above)
Comment (okanji) [reply app/(product)/private-cloud/(product)/product/[licencePlate]/@decision/page.tsx:61]: Ah yes! Will refactor

File: components/tabs/ProductsRequestsTabs.tsx (hunk 6 above)
Comment (okanji) [reply components/tabs/ProductsRequestsTabs.tsx:21]: Yes I will make that tailwind for consistency.

File: app/(product)/public-cloud/(product)/history/page.tsx (hunk 4 above)
Comment (okanji) [reply app/(product)/public-cloud/(product)/history/page.tsx:11]: I will remove that

File: app/(product)/public-cloud/(product)/product/[licencePlate]/@decision/page.tsx (hunk 5 above)
Comment (okanji) [reply app/(product)/public-cloud/(product)/product/[licencePlate]/@decision/page.tsx:68]: Yes I will use that

File: components/nav/ProfileDropdown.tsx [hunk 7]
```diff
@@ -12,6 +12,9 @@ import fetchUserImage from '@/components/nav/generateAvatar';
 export default function ProfileDropdown() {
   const { data: session, status } = useSession();
   const email = session?.user?.email;
+  const roles = session?.user?.roles;
+
+  console.log('ROLES ', roles);
```
Comment (Kolezhanchik) [inline components/nav/ProfileDropdown.tsx:17]: is this console.log needed?

Comment (okanji) [reply components/nav/ProfileDropdown.tsx:17]: No, will remove. Thanks!

File: app/(product)/public-cloud/(product)/history/[licencePlate]/page.tsx [hunk 8]
```diff
@@ -1,58 +1,61 @@
...
-
-  const { data, total } = await publicCloudRequestsPaginated(
+  const { data, total }: { data: PublicCloudRequest[]; total: number } = await publicCloudRequestsPaginated(
     +pageSize || defaultPageSize,
     currentPage,
-    search,
+    params.licencePlate,
     ministry,
-    provider,
+    cluster,
```
Comment (Kolezhanchik) [inline app/(product)/public-cloud/(product)/history/[licencePlate]/page.tsx:58]: is cluster also used for public cloud?

Comment (okanji) [reply app/(product)/public-cloud/(product)/history/[licencePlate]/page.tsx]: Will change that

**Review by Kolezhanchik — APPROVED** [review]

**Pull request merged**